src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3500,6 +3500,14 @@ protected override void VisitStatement(BoundStatement statement)
             return null;
         }
 
+        public override BoundNode? VisitUnconvertedCollectionExpression(BoundUnconvertedCollectionExpression node)
+        {
+            // This method is only involved in method inference with unbound lambdas.
+            var result = base.VisitUnconvertedCollectionExpression(node);
```

Comment:
Why are we actually visiting here, when, for example, the `VisitUnconvertedObjectCreationExpression` does not visit? I thought we would visit the contents of this expression after it is converted.

Reply:
Thanks. I think you are correct, we could skip visiting the elements here since the elements should be visited when the unbound lambda (and, as a result, the collection expression) is converted. For now though, I'll stick with the safer change to visit elements here, as we were doing before this change. (For reference, it looks like `VisitUnconvertedInterpolatedString()` also visits the sub-expressions.)

Reply:
I am curious if the decision to visit sub-expressions could improve the nullable warnings in error scenarios. But, I also don't find the question interesting enough to force further revisions on the PR in order to test it.